controllers/configuration_strat_merge_patch.go

```diff
@@ -0,0 +1,120 @@
+package controllers
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"os"
+	"path/filepath"
+
+	securejoin "github.com/cyphar/filepath-securejoin"
+	"github.com/fluxcd/pkg/http/fetch"
+	generator "github.com/fluxcd/pkg/kustomize"
+	"github.com/fluxcd/pkg/tar"
+	"github.com/open-component-model/ocm-controller/api/v1alpha1"
+	"gopkg.in/yaml.v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kustypes "sigs.k8s.io/kustomize/api/types"
+)
+
+// the following is influenced by https://github.com/fluxcd/kustomize-controller
+func (m *MutationReconcileLooper) strategicMergePatch(ctx context.Context, spec v1alpha1.MutationSpec, resource []byte, tmpDir string) (string, v1alpha1.Identity, error) {
+	source, err := m.getSource(ctx, spec.PatchStrategicMerge.Source.SourceRef)
+	if err != nil {
+		return "", nil, err
+	}
+
+	workDir, err := securejoin.SecureJoin(tmpDir, "work")
+	if err != nil {
+		return "", nil, err
+	}
+
+	dirPath, err := securejoin.SecureJoin(workDir, spec.PatchStrategicMerge.Source.Path)
+	if err != nil {
+		return "", nil, err
+	}
+
+	gzipSnapshot := &bytes.Buffer{}
+	gz := gzip.NewWriter(gzipSnapshot)
+	if _, err := gz.Write(resource); err != nil {
+		return "", nil, err
+	}
+
+	if err := gz.Close(); err != nil {
+		return "", nil, err
+	}
+
+	if err := tar.Untar(gzipSnapshot, workDir); err != nil {
+		return "", nil, err
+	}
+
+	tarSize := tar.UnlimitedUntarSize
+	fetcher := fetch.NewArchiveFetcher(10, tarSize, tarSize, "")
+	err = fetcher.Fetch(source.GetArtifact().URL, source.GetArtifact().Checksum, workDir)
+	if err != nil {
+		return "", nil, err
+	}
```

Comment:
So the content is guaranteed to be tar.gz? Should we, perhaps, use AutoDecompress? Or something akin to that?

Reply:
Hmm, the content is from a snapshot so it is guaranteed to be tar.gz right?

Reply:
Ehh, no idea. I thought we just tar that and then the retrieval does an Uncompress and then you just have to untar-it. No idea. :D The whole thing is pretty confusing. :D

Reply:
Yes that makes sense. Here t he flux `tar.Untar` expects a tar.gz, so we gzip the snapshot and then pass it to `tar.Untar`.

Reply:
Okay, so, where do you gzip it?

Reply:
L37-41

Reply:
Oh shit, I misread that. :DDD Aaaa.

So that's interesting. :D So you gzip it first, then untar which expect a gzip. This is suuuuuppppeeeer confusing. :D

Reply:
I wonder, since we do autodecompress now-a days everywhere, if we should just bite the bullet and put back the gzip into our tar and snapshots. Anyways, this is fine for now. :)

Reply:
tar.Untar expects a tar.gz. Our snapshot is a tar. So we compress and pass to the untar. We could just write another untar func?